src/pages/Facility/settings/specimen-definitions/SpecimenDefinitionForm.tsx

```diff
@@ -411,42 +401,53 @@ export function SpecimenDefinitionForm({
-                              <ValueSetSelect
-                                system="system-prepare_patient_prior_specimen_code"
-                                placeholder={t("select_patient_preparation")}
-                                onSelect={(code) =>
-                                  handlePatientPreparationSelect(code, index)
+                      {fields.map((field, index) => (
+                        <div key={field.id} className="flex items-center gap-2">
+                          <FormControl>
+                            <ValueSetSelect
+                              system="system-prepare_patient_prior_specimen_code"
+                              placeholder={t("select_patient_preparation")}
+                              value={field}
+                              onSelect={(code) => {
+                                const current = form.getValues(
+                                  "patient_preparation",
+                                );
+                                const isDuplicate = current.some(
+                                  (prep, i) =>
+                                    prep?.code === code.code && i !== index,
+                                );
+                                if (!isDuplicate) {
+                                  update(index, code);
+                                } else {
+                                  toast.error(
+                                    t("duplicate_patient_preparation"),
+                                  );
                                 }
-                                value={preparation}
-                                disabled={isLoading}
-                              />
-                            </FormControl>
-                            {field.value.length > 1 && (
-                              <Button
-                                type="button"
-                                variant="ghost"
-                                size="icon"
-                                onClick={() => removePatientPreparation(index)}
-                                className="h-10 w-10"
-                              >
-                                <XCircle className="h-5 w-5" />
-                              </Button>
-                            )}
-                          </div>
-                        ),
-                      )}
+                              }}
+                              disabled={isLoading}
+                            />
+                          </FormControl>
+                          {fields.length > 1 && (
+                            <Button
+                              type="button"
+                              variant="ghost"
+                              size="icon"
+                              onClick={() => remove(index)}
+                              className="size-10"
+                            >
+                              <XCircle className="h-5 w-5" />
```

Comment:
Do it at every place in this file 

```suggestion
                              <XCircle className="size-5" />
```